Nひきの犬をとってこれていることは確認済み

s-hayashi1104 opened the pull request:
テストコードがどうもいまいち。。。
テストコードのstep 16 のcallback.mock.calls.lengthの意味を理解できていないのかも。。。

s-hayashi1104 committed:
Nひきの犬をとってこれていることは確認済み
テストコードがどうもいまいち。。。
テストコードのstep 16 のcallback.mock.calls.lengthの意味を理解できていないのかも。。。

Mitsuru53 commented on getNRandomDogs.test.js:
```diff
@@ -0,0 +1,29 @@
+      done();
+    };
+    let callback = jest.fn(runAssertion);
+    getNRandomDogs(callback);
```
getNRandomDogsの定義は引数2つで、
テストコードでは1つしか受け取っていないですが、
こちらは書き漏れでしょうか？
```
const getNRandomDogs = function(randomInt, callback) {
  const resultArray = [];
  for (let loop = 0; loop < randomInt; loop+=1){ 
    getRandomDog(function(dog){
      resultArray.push(dog);
      callbackResultArray(randomInt, callback, resultArray);
    });
  }
};
```

仮にgetNRandomDogs(randomInt , callback)として呼び出してみてください
どういう値を期待(expect)しているが、実際はこうだよってことを示してくれます

s-hayashi1104 committed:
testコードに引数が不足していたため追加
テストコード修正

clonable-eden commented on getNRandomDogs.test.js:
```diff
@@ -0,0 +1,29 @@
+  });
+  test('should call the callback with N dogs as an argument', (done) => {
+    getNRandomDogs(randomInt, function(dogs) {
+      expect(isDog(dogs[0])).toBe(true);
```
ここは、0番目のdogだけではなく、全ての要素がdogであることをテストしなければなりません。

clonable-eden commented on getNRandomDogs.test.js:
```diff
@@ -0,0 +1,29 @@
+  test('should call the callback with N dogs as an argument', (done) => {
+    getNRandomDogs(randomInt, function(dogs) {
+      expect(isDog(dogs[0])).toBe(true);
+      expect(measureLength(dogs)).toBe(true);
```
`measureLength`という関数を定義せずとも
```javascript
expect(dogs.length).toBe(randomInt);
```
でよいのでは？

clonable-eden commented on getNRandomDogs.test.js:
```diff
@@ -0,0 +1,29 @@
+describe('getNRandomDogs', () => {
+  test('should take a callback as an argument and call it once', (done) => {
+    const runAssertion = function() {
+      expect(callback.mock.calls.length).toBe(1);
```
~~~`toBe(1)`だと1回呼ばれることをテストしているので、`randomInt`回の間違いでは？~~~

s-hayashi1104 replied:
@clonable-eden 
callbackを呼んでいるのは一回だけなのでrandomIntにするとテスト通らないですね。。。
メインの処理書き直したほうが良いでしょうか？？

clonable-eden replied:
あーそうか、失礼、テストタイトルを見逃してました。
`should take a callback as an argument and call it once`
このままで大丈夫です（このコメントは無視してください！

clonable-eden reviewed (changes requested):
@s-hayashi1104 コメントしました！

s-hayashi1104 committed:
犬を全数テストするように修正
lengthとrandomIntの比較をする処理を修正
もう一点の指摘事項に関してですが、toBe(randomInt)にするとテストとおらない
 => メインのロジック修正が必要？？

clonable-eden commented on getNRandomDogs.test.js:
```diff
@@ -0,0 +1,27 @@
+    getNRandomDogs(randomInt, function(dogs) {
+      for (let loop = 0; loop < randomInt; loop+=1) {
+        expect(isDog(dogs[loop])).toBe(true);
+      }
```
実はこの部分いい書き方があって、
```javascript
expect(dogs.every(dog => isDog(dog))).toBe(true);
```
とすることができます。
詳細はこちらを確認してください。
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/every

clonable-eden reviewed (commented):
@s-hayashi1104 LGTMです！
一箇所コメントさせてもらったので確認お願いします。

s-hayashi1104 committed:
for文からeveryに改修

clonable-eden commented:
@s-hayashi1104 LGTMです！
